### PR TITLE
feat: Send mention

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

updated `MessageComposerInnerState` to request mention-suggestions when it's needed and store mentions when user picked some.
Also added tests for it.

https://user-images.githubusercontent.com/6539347/199734440-dac27b43-766b-49e6-bf19-01b580b414cb.mp4

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1138/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4849590)(4849590)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4849590)(4849590)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1138/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4849720)(4849720)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4849720)(4849720)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1138/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4850015)(4850015)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4850015)(4850015)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1138/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4852835)(4852835)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4852835)(4852835)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageActionsBox.kt [hunk 1]
```diff
@@ -113,12 +116,20 @@ private fun AddGifAction() {
...
 
 @Composable
-private fun AddMentionAction() {
+private fun AddMentionAction(isSelected: Boolean, addMentionAction: () -> Unit) {
+    val onButtonClicked = remember(isSelected) {
+        if (isSelected) {
+            {} // we don't want to start mention flow when it's already "in progress"
+        } else {
+            addMentionAction
+        }
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageActionsBox.kt]: can't we just keep the else ?

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposerInnerState.kt [hunk 2]
```diff
@@ -115,6 +180,109 @@ data class MessageComposerInnerState(
...
+            AnnotatedString.Range(mentionSpanStyle, mention.start, mention.start + mention.length)
+        }
+
+//        return text.copy(
+//            annotatedString = AnnotatedString(
+//                text.annotatedString.text,
+//                spanStyles,
+//                text.annotatedString.paragraphStyles
+//            )
+//        )
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposerInnerState.kt:199]: Can we remove it ?

**Review by ohassine — APPROVED** [review]
It looks amazing ✨, nicely done 👌..just left some tiny comments about possible cleanup

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1138/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4858289)(4858289)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4858289)(4858289)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposerInnerState.kt (hunk 2 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposerInnerState.kt:199]: as i commented above there is an issue in Compose lib that does not allow to change `SpanStyle` for `EditTextView`. 
I kept this code commented so in a future when (if) compose-issue will be fixed we can just uncomment the code, without any additional actions :)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1138/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4858403)(4858403)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4858403)(4858403)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1138/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4858582)(4858582)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4858582)(4858582)-dev-release.apk)

**Pull request merged**